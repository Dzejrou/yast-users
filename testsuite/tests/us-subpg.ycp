/**
 * Author:	Martin Vidner <mvidner@suse.cz>
 * Summary:	test UserSettings::SortUsersByPrimaryGroup
 * $Id$
 */
{
    include "testsuite.ycp";
    import "UserSettings";

    UserSettings::current_users = SCR::Read (.target.ycp, "./tests/users.ycp.out");
    UserSettings::current_groups = SCR::Read (.target.ycp, "./tests/groups.ycp.out");

    TEST (``( UserSettings::SortUsersByPrimaryGroup () ), [], nil);

    map group = UserSettings::current_groups[100]:$[]; // users
    map prim = UserSettings::SortUsersByPrimaryGroup ();
    TEST (``( UserSettings::FindUsersBelongGroupHint (group, prim) ), [], nil);

    TEST (``( UserSettings::SortGroupsByUsers () ), [], nil);

    map gbu = UserSettings::SortGroupsByUsers ();
    TEST (``( UserSettings::FindGroupsBelongUserHint ("root", gbu) ), [], nil);
}
