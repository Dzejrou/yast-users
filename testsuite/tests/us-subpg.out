Return	$[0:[0], 1:[1], 2:[2, 26, 60, 64], 3:[62], 6:[37], 7:[4], 12:[8, 96], 13:[9], 14:[10, 33, 49], 15:[50], 25:[25], 28:[28], 31:[41], 44:[44], 49:[40], 51:[51], 54:[59], 56:[61], 60:[68], 61:[69], 62:[13], 65:[71], 70:[76], 100:[12, 36, 66, 67, 70, 11018, 11029], 65533:[65534], 65534:[30, 31, 34, 39, 65, 74]]
Return	adabas,db4web,games,jsrain,mvidner,pop,wnn
Return	$["at":["at"], "bin":["bin"], "cyrus":["mail"], "daemon":["bin"], "dpbox":["localham"], "fax":["uucp"], "fnet":["uucp"], "gdm":["shadow"], "mvidner":["video"], "named":["named"], "news":["news"], "nobody":["nobody", "nogroup"], "oracle":["dba"], "perforce":["perforce"], "postfix":["postfix"], "root":["root", "bin", "uucp", "shadow", "dialout", "audio", "nobody", "nogroup"], "sapdb":["sapdb"], "uucp":["uucp"]]
Return	root,bin,uucp,shadow,dialout,audio,nobody,nogroup
